src/ui/print.jl

```diff
@@ -34,18 +36,24 @@ The configurations below will be active whenever `show`ing [JET's analysis resul
 - `print_inference_success::Bool = true` \\
   If `true`, print a message when there is no errors found in abstract interpretation based analysis pass.
 ---
+- `stacktrace_types_limited::Bool = true` \\
+  If `true`, limit printing depth of argument types in stack traces.
+---
 """
 struct PrintConfig
     print_toplevel_success::Bool
     print_inference_success::Bool
     fullpath::Bool
-    function PrintConfig(; print_toplevel_success::Bool  = false,
+    stacktrace_types_limited::Bool
+    function PrintConfig(; print_toplevel_success::Bool = false,
                            print_inference_success::Bool = true,
-                           fullpath::Bool                = false,
+                           fullpath::Bool = false,
+                           stacktrace_types_limited::Bool = true,
```

Comment:
```suggestion
                           stacktrace_types_limit::Int = 2,
```
What if instead we made this an Int, and then users can customize how deep the type printing goes? We could document that `stacktrace_types_limit = -1` or `Inf` prints all the type information

Reply:
That said, I found 2 to be pretty reasonable, but perhaps that's not best for everyone?

Reply:
I checked out how `type_depth_limit` is implemented, and it looks like when `maxdepth` is `nothing`, the depth is adjusted dynamically based on the display size. I think this flexible depth might work better in most cases, but what do you think? It might be reasonable to make `stacktrace_types_limited::Union{Nothing,Int}=nothing`.

Reply:
That sounds fine to me, so long it's toggle-able in some way

Reply:
Alright, could you make that change if you have time? Otherwise I will work on it tomorrow.

Reply:
Will do, thanks!